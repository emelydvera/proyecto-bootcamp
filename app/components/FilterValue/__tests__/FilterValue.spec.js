const React = require('react');
const FilterValue = require("../FilterValue");
const { render, screen, fireEvent } = require('@testing-library/react');
const { I18n, I18nProvider } = require("nordic/i18n");
const { getByText } = screen;
const { data } = require('../../../../mocks/test/get/https/api.mercadolibre.com/sites/MCO/search?limit=10&offset=0&page=1&q=tablet.json');
const UrlGenerator = require('../../../utils/urlGenerator');


const i18n = new I18n({
    translations: {}
})

describe('Component <FilterValue/>', () => {

    let component;
    const props = {
        id: data.available_filters.id,
        value: data.available_filters[0].values[0],
        urlGenerator: new UrlGenerator()
    }

    beforeEach(() => {

        component = render(
            <I18nProvider i18n={i18n}>
                <FilterValue {...props} />
            </I18nProvider>
        )

    })

    it('should render the components', () => {

        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();

    });

    it('should call setFilter function on click', () => {

        const urlGenerator= new UrlGenerator();
        const {setFilter } = urlGenerator
        console.log(urlGenerator);
        const p = getByText(/Todas las tiendas oficiales/i);
        fireEvent.click(p);
        expect(setFilter).toBeInstanceOf(Function);

    });

})