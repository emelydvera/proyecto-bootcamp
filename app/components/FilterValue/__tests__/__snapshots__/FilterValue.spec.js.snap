// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <FilterValue/> should render the components 1`] = `
<DocumentFragment>
  <p
    class="andes-typography filter__section__value andes-typography--type-body andes-typography--size-m andes-typography--color-secondary"
  >
    Todas las tiendas oficiales  
    <span
      class="filter__section__value__results"
    >
      (66)
    </span>
  </p>
</DocumentFragment>
`;
